src/components/TodoList/TodoList.tsx

```diff
@@ -1,224 +1,115 @@
+    setActiveTodoId(todo.id);
+    setModalOpen(true);
+  };
+
+  const filteredTodos = todos.filter((todo: Todo) => {
+    const matchesQuery = todo.title.toLowerCase().includes(query.toLowerCase());
+
+    if (status === 'all') {
+      return matchesQuery;
+    } else if (status === 'active') {
+      return matchesQuery && !todo.completed;
+    } else if (status === 'completed') {
+      return matchesQuery && todo.completed;
+    }
+
+    return false;
+  });
 
-            <td className="has-text-right is-vcentered">
-              <button data-cy="selectButton" className="button" type="button">
+  return (
+    <>
+      {filteredTodos.length > 0 ? (
+        <table className="table is-narrow is-fullwidth">
+          <thead>
+            <tr>
+              <th>#</th>
+              <th>
                 <span className="icon">
-                  <i className="far fa-eye" />
+                  <i className="fas fa-check" />
                 </span>
-              </button>
-            </td>
-          </tr>
-        </tbody>
-      </table>
+              </th>
+              <th>Title</th>
+              <th> </th>
+            </tr>
+          </thead>
+
+          <tbody>
+            {filteredTodos.map(todo => {
+              const { title, id, completed } = todo;
+
+              return (
+                <tr data-cy="todo" key={id}>
+                  <td className="is-vcentered">{id}</td>
+                  <td className="is-vcentered">
+                    {completed ? (
+                      <span className="icon" data-cy="iconCompleted">
+                        <i className="fas fa-check" />
+                      </span>
+                    ) : (
+                      ''
+                    )}
+                  </td>
+
+                  <td className="is-vcentered is-expanded">
+                    <p
+                      className={classNames({
+                        'has-text-success': completed === true,
+                        'has-text-danger': completed === false,
+                      })}
+                    >
+                      {title}
+                    </p>
+                  </td>
+
+                  <td className="has-text-right is-vcentered">
+                    <button
+                      data-cy="selectButton"
+                      className="button"
+                      type="button"
+                      onClick={() => handleTodoClick(todo)}
+                    >
+                      <span className="icon">
+                        <i
+                          className={classNames('far', {
+                            'fa-eye-slash': modalOpen && activeTodoId === id,
+                            'fa-eye': !(modalOpen && activeTodoId === id),
+                          })}
+                        />
+                      </span>
+                    </button>
+                  </td>
+                </tr>
+              );
+            })}
```

Comment:
The `map` function inside the render method creates a new function for each item in the `filteredTodos` array. This could be a potential performance issue if the `filteredTodos` list is large. Consider moving this logic outside of the render method or use `useCallback` hook.